packages/espressocash_app/lib/features/conversion_rates/src/bl/token_price_cache.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:drift/drift.dart';
+import 'package:injectable/injectable.dart';
+
+import '../../../../data/db/db.dart';
+import '../../../../data/db/mixins.dart';
+import 'conversion_rates_client.dart';
+
+const _maxAge = Duration(minutes: 1);
+
+@injectable
+class TokenPriceCache {
+  TokenPriceCache({
+    required MyDatabase db,
+  }) : _db = db;
+
+  final MyDatabase _db;
+
+  Future<Map<String, PricesMapDto>> get(
+    List<String> ids,
+  ) async {
+    final Map<String, PricesMapDto> result = {};
+
+    for (final id in ids) {
+      final query = _db.select(_db.tokenPriceCacheRows)
+        ..where((p) => p.id.equals(id));
+
+      final row = await query.getSingleOrNull();
+
+      if (row != null) {
+        final now = DateTime.now();
+
+        if (now.difference(row.created) <= _maxAge) {
```

Comment:
Instead of retrieving and then comparing the age, you can add the condition to the query to select only records after specific DateTime. If nothing is found, just ignore it, it's not even necessary to remove the old record since it will be overwritten anyway.